src/frontend/components/Felleskomponenter/SkjemaModal/SkjemaModal.tsx

```diff
@@ -56,39 +44,43 @@ const SkjemaModal: React.FC<{
     onAvbrytCallback,
     submitKnappSpråkId,
     onSubmitCallback,
+    tittel,
+    submitKnappTekst,
     children,
 }) => {
-    const { formatMessage } = useIntl();
-
     return (
-        <StyledModal
-            isOpen={erÅpen}
-            contentLabel={formatMessage({ id: modalTittelSpråkId })}
-            onRequestClose={() => {
+        <Modal
+            open={erÅpen}
+            onClose={() => {
                 toggleModal();
                 onAvbrytCallback && onAvbrytCallback();
             }}
-            /* aria-modal blir satt til true så vi trenger ikke å gjøre aria-hidden på appen */
-            ariaHideApp={false}
         >
-            <form>
-                <StyledInnholdstittel>
-                    <SpråkTekst id={modalTittelSpråkId} />
-                </StyledInnholdstittel>
-                {children}
-                <StyledButton
-                    variant={valideringErOk() ? 'primary' : 'secondary'}
-                    type={'submit'}
-                    loading={!!submitSpinner}
-                    onClick={event => {
-                        event.preventDefault();
-                        onSubmitCallback();
-                    }}
-                >
-                    <SpråkTekst id={submitKnappSpråkId} />
-                </StyledButton>
-            </form>
-        </StyledModal>
+            <ModalContent>
+                {modalTittelSpråkId && (
+                    <StyledInnholdstittel>
+                        <SpråkTekst id={modalTittelSpråkId} />
+                    </StyledInnholdstittel>
+                )}
+                {tittel}
+                <form>
+                    {children}
+                    <StyledButton
+                        variant={valideringErOk() ? 'primary' : 'secondary'}
+                        type={'submit'}
+                        data-testid={'submit-knapp-i-modal'}
```

Comment:
Legger in testid når jeg toucher bort i tester som tidligere har brukt hardkodede tekster for å selektere elementer.